packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -636,4 +645,40 @@ export class ActionExecutionContext
 	async hackGetMediaObjectDuration(mediaId: string): Promise<number | undefined> {
 		return getMediaObjectDuration(this._context, mediaId, this.studioId)
 	}
+
+	async listPeripheralDevices(): Promise<PeripheralDevicePublic[]> {
+		const selector: Partial<PeripheralDevice> = {}
+
+		if (this._context.studio.organizationId) {
+			selector.organizationId = this._context.studio.organizationId
+		} else {
+			selector.organizationId = null
+		}
+
+		const devices = await this._context.directCollections.PeripheralDevices.findFetch(selector)
+
+		return devices.map((d) => {
+			// Only expose a subset of the PeripheralDevice to the
+			return literal<PeripheralDevicePublic>({
+				_id: d._id,
+				name: d.name,
+				studioId: d.studioId,
+				category: d.category,
+				type: d.type,
+				subType: d.subType,
+				parentDeviceId: d.parentDeviceId,
+				created: d.created,
+				status: d.status,
+				settings: d.settings,
```

Comment:
Wouldn't it make sense to expose `deviceName` from `PeripheralDevice` (https://github.com/nrkno/sofie-core/blob/031e3e3da71a9dc0eb6c4b15215eb375fc5e08ec/packages/corelib/src/dataModel/PeripheralDevice.ts#L31) and the available TSR Actions (from `configManifest`)? I think it would greatly improve on the problem highlighed by @Julusian above.

Reply:
Had a go at this, it's a little messy so suggestions on cleaning that up welcome!